drivers/serial/uart_nrfx_uarte.c

```diff
@@ -436,12 +434,9 @@ static void uarte_disable_locked(const struct device *dev, uint32_t dis_mask)
 	}
 #endif
 
-#ifdef CONFIG_SOC_NRF54H20_GPD
-	const struct uarte_nrfx_config *cfg = dev->config;
-
-	nrf_gpd_retain_pins_set(cfg->pcfg, true);
-#endif
 	nrf_uarte_disable(get_uarte_instance(dev));
+
+	(void)pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
```

Comment:
Don't we need to move this up to avoid glitching the pin? Assuming UART tx is idle high then this might disable the uart (release pin from uart) and then drive it high here if PINCTRL_STATE_SLEEP is high